src/puffin/src/file_format/reader.rs

```diff
@@ -26,10 +26,12 @@ use crate::file_metadata::FileMetadata;
 pub trait PuffinSyncReader<'a> {
     type Reader: std::io::Read + std::io::Seek;
 
-    /// fetch the FileMetadata
+    /// Fetches the FileMetadata.
```

Comment:
Nonblocking: I think maybe `PuffinReader` and `AsyncPuffinReader` are clearer names.